packages/openshift/context.py

```diff
@@ -319,7 +320,11 @@ def register_action(self, action):
 
             if c.tracking_strategy:
                 if isinstance(c.tracking_strategy, Result):
-                    c.tracking_strategy.add_action(action)
+                    if(len(c.tracking_strategy.actions())<c.limit_tracking_actions):
```

Comment:
- For coding style, don't include outer parens for expressions that don't require it (see if statements that precede this one)
- I believe this logic is flawed when limit_tracking_actions is None.
Keep DRY when possible:
```
add_action(action)
if limit is not None and len(actions) > limit:
    pop
```